src/ConfigManager.cpp

```diff
@@ -23,32 +23,41 @@ void ConfigManager::setup() {
   DebugPrint(F("MAC: "));
   DebugPrintln(WiFi.macAddress());
 
-  DebugPrintln(F("Reading saved configuration"));
+  DebugPrintln(F("Checking for magic initialization"));
   EEPROM.get(0, magic);
 
   if (memcmp(magic, magicBytes, MAGIC_LENGTH) == 0) {
-    EEPROM.get(MAGIC_LENGTH, ssid);
-    DebugPrint(F("SSID: \""));
-    DebugPrint(ssid);
-    DebugPrintln(F("\""));
-
-    EEPROM.get(MAGIC_LENGTH + SSID_LENGTH, password);
+    DebugPrintln(F("Reading saved configuration"));
     readConfig();
 
-    WiFi.begin(ssid, password[0] == '\0' ? NULL : password);
+    EEPROM.get(MAGIC_LENGTH, ssid);
 
-    if (wifiConnected()) {
-      DebugPrint(F("Connected to "));
+    if (strlen(ssid) > 0) {
+      DebugPrint(F("SSID: \""));
       DebugPrint(ssid);
-      DebugPrint(F(" with "));
-      DebugPrintln(WiFi.localIP());
+      DebugPrintln(F("\""));
+
+      EEPROM.get(MAGIC_LENGTH + SSID_LENGTH, password);
+      WiFi.begin(ssid, password[0] == '\0' ? NULL : password);
+
+      if (wifiConnected()) {
+        DebugPrint(F("Connected to "));
+        DebugPrint(ssid);
+        DebugPrint(F(" with "));
+        DebugPrintln(WiFi.localIP());
 
-      WiFi.mode(WIFI_STA);
+        WiFi.mode(WIFI_STA);
 
-      startApi();
+        startApi();
+      }
+    } else {
+      DebugPrintln(F("No SSID found"));
     }
   } else {
     // We are at a cold start, don't bother timing out.
+    if (initCallback) {
```

Comment:
When the magic is not set, the `initCallback` is called which can do things .. current use case is set some config defaults